### ISSUE TITLE
[Bug]: S3 pipefiles in docker container has no permission

### ISSUE THREAD

**Issue by bergo** [drakkan/sftpgo#1441]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.

### Bug description

The pipefile for write operations against S3 are written in the root of the container. There are no permissions for the sftpgo user inside the container. Even the setting of the temp_path does not affect the location where the pipefiles are written. 

(PS: if the / has write or owner permission, it works of course)

### Steps to reproduce

1. Using SFTPGo inside a docker container as described. 
2. Configured the S3 folder or user as described. 
3. Read operations are working
4. All write operations are not working
5. Setting "/" in docker container with group or write permission, works

### Expected behavior

The write operations should work.

### SFTPGo version

2.5.4

### Data provider

sqlite/s3

### Installation method

Community Docker image

### Configuration

# Proxy Protocol
SFTPGO_COMMON__PROXY_PROTOCOL=2

# with or without
SFTPGO_COMMON__TEMP_PATH="/srv/sftpgo/"

# Proxy Protocol
SFTPGO_COMMON__PROXY_ALLOWED=0.0.0.0

# SFTP
SFTPGO_SFTPD__BINDINGS__0__PORT=1111
SFTPGO_SFTPD__BINDINGS__0__PROXY_ALLOWED=0.0.0.0

....

### Relevant log output

```shell
{"level":"error","time":"2023-10-23T06:31:52.245","sender":"SFTP","connection_id":"SFTP_64f101c31484e398c735b7b1b22c65512e61c30502a1173b8be39df6b30f9900_1","message":"could not open file \"raw/xxx/yyy.jpg\" for reading: open /pipefile2416887562: permission denied"}
{"level":"error","time":"2023-10-23T06:31:52.245","sender":"SFTP","connection_id":"SFTP_64f101c31484e398c735b7b1b22c65512e61c30502a1173b8be39df6b30f9900_1","message":"generic path error: open /pipefile2416887562: permission denied"}
```

### What are you using SFTPGo for?

Private user, home usecase (home backup/VPS)

**Comment by drakkan** [comment]
This is just a support request, not a bug. Please check our support policy. Thanks for understanding

**Comment by bergo** [comment]
@drakkan thanks for your answer and work. But why is this a "support request"? It is not working for the default configuration as described. I need no support, i just reported it that you know that it is not working ;)

**Comment by halbgut** [comment]
The user's configured "Home Dir" is used to store temporary files in. I wrongly put `/` in the "Home Dir" input, which caused this same issue. SFTPGo helpfully sets it to `/tmp/[username]` by default, so best use that instead.

If you set it to a path that the `sftpgo` user. I ran into the same issue while setting up SFTPGo. Looking at the docs I thought that `common.temp_path` would be used as a location of the temporary paths, but looking at the source code I figured out that it was using the user's home directory instead.

Leaving this here in case anyone else stumbles on this.

**Comment by gustaff-weldon** [comment]
I had the same issue. I have setup `/transfers` as writeable volume and set it as `common.temp_path`, but I was still getting errors to like `error: open /upload/pipefile2227123473: no such file or directory` when uploading to GCS hosted folder - upload to filesystem worked fine even before.

According to [`temp_path ` docs ](https://docs.sftpgo.com/2.7/config-file/#common):
> Defines the path for temporary files such as those used for atomic uploads or file pipes. If you set this option you must make sure that the defined path exists, is accessible for writing by the user running SFTPGo

@drakkan either it is a bug with `temp_path` not working, or documentation is lacking some crucial details.

I have also tried setting just user root dir aka `home_dir` to that `/transfers`. That did not work either.
I had to set **both** `home_dir` and virtual folder `mapped_path`  to `/transfers` to make uploads to GCS work.

I originally had set the `/mapped_path` to a folder within bucket that user have writeable access to, but it did not work.

Also, docs for virtual folder do not mention mapped path, they refer to `virtual_path`. Virtual folder from in WebAdmin has a field called `Root directory` which is saved as `mapped_path` on virtual folder 🤷 so it is a bit confusing what is what.

Even with `/transfers` folder fix, I still cannot make uploads to stream to bucket with `upload_mode: 0`.
